xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ExportAction.java

```diff
@@ -136,52 +225,100 @@ private String exportHTML(XWikiContext context) throws XWikiException, IOExcepti
         return null;
     }
 
-    private Collection<DocumentReference> resolvePagesToExport(String[] pages, XWikiContext context)
+    private String extractWikiName(String pattern, XWikiContext context)
+    {
+        String wikiName;
+        if (pattern.contains(":")) {
```

Comment:
You should do the if on `index` instead of parsing the String twice (only substring if index > 0).